sw/device/mask_rom/boot.md

```diff
@@ -209,63 +451,151 @@ crt_init() {
 
-```c
-manufacturing_boot_strap() {
-  // 1. Read lifecycle state to establish whether bootstrapping is allowed.
-  // 2. Determine what kind of bootstrapping is being requested.
-  // 3. Load ROM_EXT image into flash if allowed.
-  // 4. Reboot device (causing loaded ROM_EXT to be verified then executed).
-}
-```
+In order to keep the Mask ROM simple, a particular Mask ROM version will only
+support one version each of the following structures. This means they must be
+carefully designed to be extensible if the other systems accessing them may
+require additional data in these formats.
 
-## Boot Info Policy
+### Key Management Data
 
-```c
-read_boot_policy() {
-  // Parameters:
-  // - initilized flash_ctrl DIF (for accessing flash info page)
-  // Returns:
-  // - boot policy struct
-  //   **Open Q:** What boot policies do we allow?
-  //   - Active ROM_EXT selector (there are only two ROM_EXT slots) (2.b, 2.c.i)
-  //   - What to do if digest doesn't match (2.c.ii)
-  //   - What to do if signature doesn't verify (2.c.iii)
-  //   - What to do if system measurements are not right (2.c.iv)
+This is the storage for the IDs of keys that are used to sign a ROM_EXT. The
+ROM_EXT image contains the full key, this stores only the key id (which can be
+computed from the full key), and uses OTPs to decide whether a key is allowed to
+be used to validate a ROM_EXT.
 
-  // 1. Uses dif_flash_ctrl to issue read of boot info page.
-  // 2. Pull this into a struct to return.
-}
-```
+Accessed by:
 
-## Lock Down Peripherals
+*   Mask ROM (to identify public key is Valid).
+*   ROM_EXT (to prevent use of public key).
 
-```c
-peripheral_lockdown() {
-  // Parameters:
-  // - ROM_EXT manifest
-  // - Handles for Peripheral DIFs
+Needs to contain:
 
-  // **Open Q:** When and How do we lock down peripheral configuration?
-  // - We configure based on ROM_EXT manifest (signed)
-  // - Only locks down these peripherals if info is provided.
+*   Read-only list of public key ids.
+*   OTP fuses to say whether a key id is revoked or not (keys start not revoked).
 
-  // What do we want to allow lockdown of?
-  // - Alert Manager
-  // - Pinmux / Padctrl
-  // - Entropy Configuration
-  // - Anything Else?
+Stored In: OTP and ROM
 
-  // This is explicitly a separate step from locking the key manager inputs. In
-  // particular, this depends on the ROM_EXT's choices, whereas the key manager
-  // inputs getting locked is not something the ROM_EXT can choose.
-}
-```
+Extensibility: None. A ROM_EXT can disable the use of a key, but cannot add new
+valid root key IDs. There will be a fixed set of root key IDs as part of a given
+Mask ROM version.
+
+### Boot Policy Structure
+
+This is the in-flash structure that defines which ROM_EXT should be booted next,
+whether it should fall back to the other ROM_EXT if it fails to boot, and/or
+whether it should be marked as the primary ROM_EXT if it succeeds to validate.
```

Comment:
Does marking a ROM_EXT primary imply programming OTP fuses or should it be a separate option?
Also, I guess there can be cases where anti-rollback blocks the secondary if it is older than the primary.

Reply:
I'm not sure. I'll add an open question about this to the part of `void boot(void)` which covers this issue.